vars/spinInfra.groovy

```diff
@@ -7,127 +7,156 @@ import uk.gov.hmcts.contino.MetricsPublisher
 import uk.gov.hmcts.pipeline.AKSSubscriptions
 import uk.gov.hmcts.contino.RepositoryUrl
 
-def call(productName, environment, tfPlanOnly, subscription) {
-  call(productName, null, environment, tfPlanOnly, subscription)
-}
-
-def call(product, component, environment, tfPlanOnly, subscription) {
-  call(product, component, environment, tfPlanOnly, subscription, "")
-}
-
-def call(product, component, environment, tfPlanOnly, subscription, deploymentTarget) {
-  def branch = new ProjectBranch(env.BRANCH_NAME)
+Closure infraArgs(Map args = [:]) {
```

Comment:
revert the last change you did and then just add a def config inside the last `def call` function you don't need this closure approach it was just the config bit that's needed